examples/name-service/programs/name-service/src/lib.rs

```diff
@@ -0,0 +1,249 @@
+use std::net::{Ipv4Addr, Ipv6Addr};
+
+use anchor_lang::prelude::*;
+use borsh::{BorshDeserialize, BorshSerialize};
+use light_hasher::{errors::HasherError, DataHasher, Discriminator, Hasher, Poseidon};
+use light_sdk::{light_accounts, verify::verify, LightDiscriminator, LightTraits};
+use light_system_program::{
+    invoke::processor::CompressedProof,
+    sdk::{
+        address::derive_address,
+        compressed_account::{
+            CompressedAccount, CompressedAccountData, PackedCompressedAccountWithMerkleContext,
+        },
+        CompressedCpiContext,
+    },
+    InstructionDataInvokeCpi, NewAddressParamsPacked, OutputCompressedAccountWithPackedContext,
+};
+use light_utils::hash_to_bn254_field_size_be;
+
+declare_id!("7yucc7fL3JGbyMwg4neUaenNSdySS39hbAk89Ao3t1Hz");
+
+#[program]
+pub mod name_service {
+
+    use light_sdk::utils::create_cpi_inputs_for_new_address;
+
+    use super::*;
+
+    pub fn create_record<'info>(
+        ctx: Context<'_, '_, '_, 'info, NameService<'info>>,
+        proof: CompressedProof,
+        new_address_params: NewAddressParamsPacked,
```

Comment:
Can the user choose the address seeds completely arbitrary?

Reply:
We could derive the address from the name to make every name unique.

Reply:
Done.